parsedmarc/__init__.py

```diff
@@ -1428,12 +1435,44 @@ def get_dmarc_reports_from_mailbox(connection: MailboxConnection,
         connection.create_folder(smtp_tls_reports_folder)
         connection.create_folder(invalid_reports_folder)
 
-    messages = connection.fetch_messages(reports_folder, batch_size=batch_size)
+    if since:
+        _since = 1440  # default one day
+        if re.match(r'\d{1,2}[mhd]$', since):
+            s = re.split(r'(\d+)', since)
+            match s[2]:
+                case 'm': _since = int(s[1])
+                case 'h': _since = int(s[1])*60
+                case 'd': _since = int(s[1])*60*24
+                case 'w': _since = int(s[1])*60*24*7
+        else:
+            logger.warning("Incorrect format for \'since\' option. \
+                           Provided value:{0}, Expected values:(5m|3h|2d|1w). \
+                           Ignoring option, fetching messages for last 24hrs"
+                           .format(since))
+
+        if isinstance(connection, IMAPConnection):
```

Comment:
Maybe add a reference to [RFC3501](https://www.rfc-editor.org/rfc/rfc3501#page-52) in order to indicate that the restriction (no time, no timezone) is on the SMTP protocol

Reply:
Done. Thanks.